populator-machinery/controller.go

```diff
@@ -590,9 +599,11 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 		// Get all ReferenceGrants in data source's namespace
 		referenceGrants, err := c.referenceGrantLister.ReferenceGrants(*dataSourceRef.Namespace).List(labels.Everything())
 		if err != nil {
+			c.prometheusMetrics.recordError("controller.ReferenceGrants.List", pvc.UID, err)
 			return fmt.Errorf("error getting ReferenceGrants in %s namespace from api server: %v", *dataSourceRef.Namespace, err)
 		}
 		if allowed, err := IsGranted(ctx, pvc, referenceGrants); err != nil || !allowed {
+			c.prometheusMetrics.recordError("IsGranted", pvc.UID, err)
```

Comment:
"IsGranted" -> "controller. IsGranted"?

Reply:
If we follow https://github.com/kubernetes-csi/lib-volume-populator/pull/172#issuecomment-2332421251 then I can just delete this.